File: TransCoder_tokenized_test_set_functions/java/LARGEST_SUBSET_GRAPH_VERTICES_EDGES_2_COLORS.java
static int subsetGraph ( int C [ ] [ ] ) {
  HashSet < Integer > vertices = new HashSet < > ( );
  for ( int i = 0;
  i < N;
  ++ i ) {
    vertices . add ( i );
  }
  while ( ! vertices . isEmpty ( ) ) {
    if ( vertices . size ( ) == 1 ) {
      return 1;
    }
    boolean someone_removed = false;
    for ( int x : vertices ) {
      HashSet < Integer > values = new HashSet < > ( );
      for ( int y : vertices ) {
        if ( y != x ) {
          values . add ( C [ x ] [ y ] );
        }
      }
      if ( values . size ( ) == 1 ) {
        vertices . remove ( x );
        someone_removed = true;
        break;
      }
    }
    if ( ! someone_removed ) {
      break;
    }
  }
  return ( vertices . size ( ) );
}

File: TransCoder_tokenized_test_set_functions/python/LARGEST_SUBSET_GRAPH_VERTICES_EDGES_2_COLORS.py
def subsetGraph ( C ) :
    global N
    vertices = set ( )
    for i in range ( N ) :
        vertices . add ( i )
    while ( len ( vertices ) != 0 ) :
        if ( len ( vertices ) == 1 ) :
            return 1
        someone_removed = False
        for x in vertices :
            values = set ( )
            for y in vertices :
                if ( y != x ) :
                    values . add ( C [ x ] [ y ] )
            if ( len ( values ) == 1 ) :
                vertices . remove ( x )
                someone_removed = True
                break
        if ( not someone_removed ) :
            break
    return len ( vertices )
